fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/BestMvSelector.java

```diff
@@ -152,15 +177,38 @@ private void calculateStatistics(OptExpression expr, OptimizerContext context) {
         expr.setStatistics(expressionContext.getStatistics());
     }
 
+    private int calcSortScore(MaterializedView mv, Set<String> equivalenceColumns, Set<String> nonEquivalenceColumns) {
+        List<Column> schema = mv.getBaseSchema();
+        int score = 0;
+        for (Column col : schema) {
+            String columName = col.getName().toLowerCase();
+            if (equivalenceColumns.contains(columName)) {
```

Comment:
Why not check if it's a short key ? Not all of columns can be the sort-keys

Reply:
change to only consider sort key